### ISSUE TITLE
Ape Escape Corrupted graphics in scene (SwanStation)

### ISSUE THREAD

**Issue by edlink98** [libretro/swanstation#76]
Here I wanted to leave a bug that I found in the Ape Escape Spanish version... I want to clarify which game is from Redump and I have the corresponding bios, I also have the .sbi file of the game. In CHD format.
This error occurs when you defeat Specter and the game credits roll, a scene appears with corrupted or glitched graphics. This happened to me in the SwanStation of Retroarch 1.15.0 Stable, (I didn't try it in the Native Duckstation). I tried it on Bettle PSX HW and it doesn't have that error... (I have all the cores updated)
Here I recorded the error and put the configuration of the two cores in the description.

https://youtu.be/Ehd2tlJkDzc

**Video description:**
Game: Ape Escape: La Invasión de los Monos
Serial: SCES-02031

Settings in Retroarch:

SwanStation (Retroarch):
Internal Resolution Scale: 5x (1080p)
Multisample Antialiasing: off
Texture Filtering: Nearest-Neighbor
PGXP Geometry Correction: off

Bettle PSX HW (Retroarch):
Internal GPU Resolution: 4x
Texture Filtering: Nearest
Multi-Sampled Anti Aliasing: 1x
PGXP Operation Mode: off

I want to share my experience of Ape Escape the Spanish version and that is that I was playing the game in Retroarch with the Swanstation core (Duckstation) and I skipped that corrupt scene, my game is Redump and it should not have any disk problem... I read that in some versions of Ape Escape it has glitches in scenes but I don't know if this would be in general... but this time I tried it and it looked horrible, I played it again but now in the retroarch Bettle PSX HW core and it works looks perfect.
I want to make it clear that I like to play with the only modification of the scaling in 1080p and nothing else.

**Comment by DarthMew** [comment]
What graphics backend are you using (Vulkan, OpenGL), and what hardware (CPU and GPU) and platform (Windows, Linux, Android etc.) are you running it on?

Also, have you tried enabling the "Use Software Renderer For Readbacks" option in SwanStation?
I know Beetle PSX has it enabled by default, and it might be worth trying to see if it helps in this case.

**Comment by edlink98** [comment]
I use vulkan
Platform: Windows 10 Pro

My Hardware:
CPU: Ryzen 5 3400g
GPU: NVIDIA 2060 12gb
Ram: 16gb

"Use Software Renderer For Readbacks" I do not have it activated. I'll try it activated and comment the results.

**Comment by edlink98** [comment]
I already did the test and activated that option and the graphic problem that that scene had was solved. Thank you! I have a query... what exactly is this option for?

Here is the test video, (https://youtu.be/FKOFqIzvrgw) it also shows the options activated when using the SwanStation... all other options are as the first video the only change is the "Use Software Renderer For Readbacks"

**Comment by DarthMew** [comment]
When the option is disabled, VRAM is being sourced entirely from the hardware renderer, which works fine for the most part until you encounter a scenario where it is missing certain graphical effects, or (in this case) produced corrupted graphics.

Enabling the option runs the software renderer in parallel, ensuring that the game has access to a more accurate source for VRAM readbacks.

**Comment by edlink98** [comment]
Wow, you learn something new every day! I have another question about another game but it will be in a new issue.
